### PR TITLE
Ossmc regression extended

### PR DESCRIPTION
**Pull request by ScriptingShrimp**
smoke testing for OCP workloads > deployments for presence of service mesh elements 

additional refactoring and naming conventions 

tested on OCP all passing

### CODE REVIEW

File: plugin/cypress/e2e/featureFiles/workloads-deployments.feature [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    Background:
+        Given user is logged as administrator in OCP Console
+        And cypress intercept hooks for workloads are registered
+        And user navigates into "k8s/ns/istio-system/deployments/kiali" page
+        And user clicks on Service Mesh tab in horizontal nav
+
+    Scenario: Verify that content of the Overview tab is correct
+        Then user clicks tab with "Overview" button
+        And user is able to see WorkloadDescriptionCard with Kiali Workload
```
Comment (ferhoyos) [inline plugin/cypress/e2e/featureFiles/workloads-deployments.feature:16]:
```suggestion
        When user clicks tab with "Overview" button
        Then user is able to see WorkloadDescriptionCard with Kiali Workload
```
Proper action here is `When`

File: plugin/cypress/e2e/featureFiles/workloads-deployments.feature [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    Scenario: Verify that content of the Traffic tab is correct
+        Then user clicks tab with "Traffic" button
```
Comment (ferhoyos) [inline plugin/cypress/e2e/featureFiles/workloads-deployments.feature:19]:
```suggestion
        When user clicks tab with "Traffic" button
```
Proper action here is `When`

File: plugin/cypress/e2e/workloads-deployments.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+When('user clicks tab with {string} button', (tabName:string) => {
+    cy.get('button').contains(tabName).click()
+});
+
+When('user clicks on Service Mesh tab in horizontal nav', () => {
+    cy.get('[data-test-id="horizontal-link-Service Mesh"]').contains('Service Mesh').click()
+});
+
+When('user is able to see WorkloadDescriptionCard with Kiali Workload', () => {
```
Comment (ferhoyos) [inline plugin/cypress/e2e/workloads-deployments.ts:19]:
```suggestion
Then('user is able to see WorkloadDescriptionCard with Kiali Workload', () => {
```
Proper action here is `Then`

File: plugin/cypress/e2e/workloads-deployments.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    cy.get('[data-test="workload-description-card"]').contains('kiali').should('be.visible')
+});
+
+When('Kiali container is selected', () => {
+    cy.get('[data-test="workload-logs-pod-containers"]').contains('kiali').click()
+});
+
+When('user sees {string} dropdown', (dropdownText:string) => {
```
Comment (ferhoyos) [inline plugin/cypress/e2e/workloads-deployments.ts:27]:
```suggestion
Then('user sees {string} dropdown', (dropdownText:string) => {
```
Proper action here is `Then`

**Review by ferhoyos — CHANGES_REQUESTED** [review]
I have not tested it but code LGTM, just small comments.

**Review by nrfox — COMMENTED** [review]
LGTM but I'll let @ferhoyos approve since he's more familiar with OSSMC

**Review by ferhoyos — APPROVED** [review]
LGTM. They are additional steps to be added when bookinfo application can be installed on tests (https://github.com/kiali/openshift-servicemesh-plugin/pull/245), but this could be done as a separate PR.

**Pull request merged**